// @looker/components/LkFieldTree/LkFieldItemContent.js
import styled from 'styled-components';
import { ListItemLabel } from '../ListItem';
import { generateIndent } from '../Tree/utils';
import { lkFieldItemDensity } from './defaults';
export var LkFieldItemContent = styled.div.attrs(function (_ref) {
  var _ref$role = _ref.role,
      role = _ref$role === void 0 ? 'treeitem' : _ref$role;
  return {
    role: role
  };
}).withConfig({
  displayName: "LkFieldItemContent",
  componentId: "sc-vth29c-0"
})(["", "{padding:0;padding-left:", ";}", " ", " background:none;color:inherit;cursor:", ";display:flex;flex:1;font-size:inherit;font-weight:inherit;margin:0;min-width:0;outline:none;position:relative;text-align:left;text-decoration:none;width:100%;&:hover,&:focus{color:inherit;text-decoration:none;}&[disabled]{color:", ";cursor:not-allowed;}"], ListItemLabel, function (_ref2) {
  var theme = _ref2.theme;
  return theme.space.xxsmall;
}, function (_ref3) {
  var depth = _ref3.depth;
  return generateIndent({
    density: lkFieldItemDensity,
    depth: depth
  });
}, function (_ref4) {
  var focusVisible = _ref4.focusVisible,
      theme = _ref4.theme;
  return focusVisible && "\n    &::after {\n      bottom: 0;\n      box-shadow: inset 0 0 0 2px ".concat(theme.colors.keyFocus, ";\n      content: '';\n      left: 0;\n      position: absolute;\n      right: 0;\n      top: 0;\n      z-index: 1;\n    }\n  ");
}, function (_ref5) {
  var cursorPointer = _ref5.cursorPointer;
  return cursorPointer ? 'pointer' : undefined;
}, function (_ref6) {
  var theme = _ref6.theme;
  return theme.colors.text1;
});
