// @looker/components/cjs/PageSize/PageSize.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.PageSize = exports.PageSizeLayout = void 0;

require("core-js/modules/es.array.map.js");

require("core-js/modules/es.array.iterator.js");

var _reactI18next = require("react-i18next");

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _Form = require("../Form");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const defaultPageSizes = [10, 25, 50, 100];

const stringToSelectOption = option => ({
  label: String(option),
  value: String(option)
});

const arrayToSelectOptions = options => options.map(option => stringToSelectOption(String(option)));

const PageSizeLayout = ({
  alwaysVisible = false,
  value,
  total,
  className,
  onChange,
  options = defaultPageSizes
}) => {
  const _useTranslation = (0, _reactI18next.useTranslation)('PageSize'),
        t = _useTranslation.t;

  const handleOnChange = newValue => onChange(Number(newValue));

  return alwaysVisible || Math.min(...options) < total ? _react.default.createElement("div", {
    className: className
  }, t('Display'), _react.default.createElement(_Form.Select, {
    width: "5rem",
    mx: "xsmall",
    options: arrayToSelectOptions(options),
    value: String(value > total ? total : value),
    onChange: handleOnChange,
    disabled: options.every(option => option > total)
  }), _react.default.createElement("span", null, t('of'), " ", total)) : null;
};

exports.PageSizeLayout = PageSizeLayout;
const PageSize = (0, _styledComponents.default)(PageSizeLayout).withConfig({
  displayName: "PageSize",
  componentId: "sc-lmdbtc-0"
})(["align-items:center;display:flex;font-size:", ";"], ({
  theme
}) => theme.fontSizes.small);
exports.PageSize = PageSize;
