// @looker/components/cjs/DataTable/Column/columnSize.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.columnSize = exports.sizeInfersTruncate = void 0;

require("core-js/modules/es.array.includes.js");

var _designTokens = require("@looker/design-tokens");

var _styledComponents = require("styled-components");

const sizeInfersTruncate = size => size && typeof size !== 'number' && !['auto', 'nowrap'].includes(size);

exports.sizeInfersTruncate = sizeInfersTruncate;
const columnSizeVariants = (0, _designTokens.variant)({
  prop: 'size',
  variants: {
    small: {
      maxWidth: '3rem',
      minWidth: '3rem'
    },
    medium: {
      maxWidth: '12rem',
      minWidth: '12rem'
    },
    large: {
      maxWidth: '20rem',
      minWidth: '20rem'
    },
    nowrap: {
      whiteSpace: 'nowrap'
    }
  }
});

const percentWidth = width => `width: ${width}%;`;

const columnSize = (0, _styledComponents.css)(["", ""], ({
  size
}) => size && typeof size === 'number' ? percentWidth(size) : columnSizeVariants);
exports.columnSize = columnSize;
