// @looker/components/Form/Inputs/Checkbox/Checkbox.js
import "core-js/modules/es.object.keys.js";
import "core-js/modules/es.array.index-of.js";
import "core-js/modules/es.symbol.js";
import "core-js/modules/es.array.is-array.js";
import "core-js/modules/es.symbol.description.js";
import "core-js/modules/es.object.to-string.js";
import "core-js/modules/es.symbol.iterator.js";
import "core-js/modules/es.array.iterator.js";
import "core-js/modules/es.string.iterator.js";
import "core-js/modules/web.dom-collections.iterator.js";
import "core-js/modules/es.array.slice.js";
import "core-js/modules/es.function.name.js";
import "core-js/modules/es.array.from.js";
import "core-js/modules/es.object.define-property.js";
import "core-js/modules/es.array.filter.js";
import "core-js/modules/es.object.get-own-property-descriptor.js";
import "core-js/modules/es.array.for-each.js";
import "core-js/modules/web.dom-collections.for-each.js";
import "core-js/modules/es.object.get-own-property-descriptors.js";
import "core-js/modules/es.object.define-properties.js";
import "core-js/modules/es.object.assign.js";
var _excluded = ["className", "checked", "defaultChecked", "onChange", "readOnly", "style", "validationType"];

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _iterableToArrayLimit(arr, i) { var _i = arr == null ? null : typeof Symbol !== "undefined" && arr[Symbol.iterator] || arr["@@iterator"]; if (_i == null) return; var _arr = []; var _n = true; var _d = false; var _s, _e; try { for (_i = _i.call(arr); !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import isUndefined from 'lodash/isUndefined';
import noop from 'lodash/noop';
import pick from 'lodash/pick';
import React, { forwardRef, useState, useEffect } from 'react';
import styled from 'styled-components';
import { reset, space } from '@looker/design-tokens';
import { mergeClassNames } from '../../../utils';
import { inputRippleColor, RIPPLE_RATIO, rippleHandlerKeys, rippleStyle, useRipple, useRippleHandlers } from '../../../Ripple';
import { pickInputProps } from '../InputProps';
import { CheckMark } from './CheckMark';
import { CheckMarkMixed } from './CheckMarkMixed';
import { FauxCheckbox } from './FauxCheckbox';
export var Checkbox = styled(forwardRef(function (props, ref) {
  var className = props.className,
      checked = props.checked,
      defaultChecked = props.defaultChecked,
      onChange = props.onChange,
      readOnly = props.readOnly,
      style = props.style,
      validationType = props.validationType,
      restProps = _objectWithoutProperties(props, _excluded);

  var _useState = useState(!!defaultChecked),
      _useState2 = _slicedToArray(_useState, 2),
      isChecked = _useState2[0],
      setIsChecked = _useState2[1];

  var _useRipple = useRipple({
    color: inputRippleColor(isChecked !== false, validationType === 'error'),
    size: RIPPLE_RATIO
  }),
      callbacks = _useRipple.callbacks,
      rippleClassName = _useRipple.className,
      rippleStyle = _useRipple.style;

  var rippleHandlers = useRippleHandlers(callbacks, _objectSpread({}, pick(restProps, rippleHandlerKeys)), restProps.disabled);
  var handleClick = readOnly ? undefined : function (e) {
    if (isUndefined(checked)) {
      setIsChecked(!isChecked);
    }

    if (onChange) {
      onChange(e);
    }
  };
  useEffect(function () {
    if (!isUndefined(checked)) {
      setIsChecked(checked);
    }
  }, [checked]);
  return React.createElement("div", _extends({
    className: mergeClassNames([className, rippleClassName]),
    style: _objectSpread(_objectSpread({}, style), rippleStyle)
  }, rippleHandlers), React.createElement("input", _extends({
    type: "checkbox"
  }, pickInputProps(restProps), {
    checked: !!isChecked,
    "aria-checked": checked,
    "aria-invalid": validationType === 'error' ? 'true' : undefined,
    onClick: handleClick,
    onChange: noop,
    ref: ref
  })), React.createElement(FauxCheckbox, {
    isSelected: !!isChecked
  }, checked === 'mixed' ? React.createElement(CheckMarkMixed, null) : React.createElement(CheckMark, null)));
})).withConfig({
  displayName: "Checkbox",
  componentId: "sc-9j2vap-0"
})(["", " ", " ", " align-items:center;display:flex;height:", ";justify-content:center;position:relative;width:", ";input{cursor:", ";height:100%;left:0;margin:0;opacity:0;position:absolute;top:0;width:100%;z-index:1;&[aria-invalid='true']{+ ", "{border-color:", ";}&:checked + ", "{background:", ";}}&:disabled{+ ", ",&:not(:checked):hover + ", "{border-color:", ";}&:checked + ", "{background:", ";}}&:not(:checked):not([aria-invalid='true']):not(:disabled){&:hover,&:focus{+ ", "{border-color:", ";}}}}"], reset, space, rippleStyle, function (_ref) {
  var space = _ref.theme.space;
  return space.u6;
}, function (_ref2) {
  var space = _ref2.theme.space;
  return space.u6;
}, function (_ref3) {
  var readOnly = _ref3.readOnly,
      disabled = _ref3.disabled;
  return readOnly || disabled ? 'not-allowed' : undefined;
}, FauxCheckbox, function (_ref4) {
  var theme = _ref4.theme;
  return theme.colors.critical;
}, FauxCheckbox, function (_ref5) {
  var theme = _ref5.theme;
  return theme.colors.critical;
}, FauxCheckbox, FauxCheckbox, function (_ref6) {
  var theme = _ref6.theme;
  return theme.colors.ui2;
}, FauxCheckbox, function (_ref7) {
  var theme = _ref7.theme;
  return theme.colors.ui2;
}, FauxCheckbox, function (_ref8) {
  var theme = _ref8.theme;
  return theme.colors.ui5;
});
