// @looker/components/Button/ButtonItem.js
var _excluded = ["children", "onClick", "value", "onBlur", "onKeyUp"],
    _excluded2 = ["focusVisible"];

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import "core-js/modules/es.array.includes.js";
import "core-js/modules/es.string.includes.js";
import "core-js/modules/es.string.small.js";
import "core-js/modules/es.object.keys.js";
import "core-js/modules/es.array.index-of.js";
import "core-js/modules/es.symbol.js";
import "core-js/modules/es.object.assign.js";

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React, { forwardRef, useContext } from 'react';
import styled from 'styled-components';
import { space, omitStyledProps } from '@looker/design-tokens';
import { useFocusVisible } from '../utils';
import { inputHeight } from '../Form/Inputs/height';
import { ButtonSetContext } from './ButtonSetContext';
var ButtonLayout = forwardRef(function (_ref, ref) {
  var children = _ref.children,
      onClick = _ref.onClick,
      value = _ref.value,
      onBlur = _ref.onBlur,
      onKeyUp = _ref.onKeyUp,
      props = _objectWithoutProperties(_ref, _excluded);

  var _useContext = useContext(ButtonSetContext),
      disabled = _useContext.disabled,
      contextValue = _useContext.value,
      onItemClick = _useContext.onItemClick;

  var _useFocusVisible = useFocusVisible({
    onBlur: onBlur,
    onKeyUp: onKeyUp
  }),
      focusVisible = _useFocusVisible.focusVisible,
      focusVisibleProps = _objectWithoutProperties(_useFocusVisible, _excluded2);

  function handleClick(e) {
    onClick && onClick(e);

    if (!e.defaultPrevented) {
      onItemClick && onItemClick(e);
    }
  }

  var itemValue = '';

  if (value !== undefined) {
    itemValue = value;
  } else if (typeof children === 'string') {
    itemValue = children;
  }

  var selected = false;

  if (contextValue) {
    if (typeof contextValue === 'string') {
      selected = contextValue === itemValue;
    } else {
      selected = contextValue.includes(itemValue);
    }
  }

  return React.createElement("button", _extends({
    "aria-pressed": selected,
    ref: ref,
    onClick: handleClick,
    value: itemValue,
    disabled: disabled,
    "data-focusvisible": focusVisible
  }, focusVisibleProps, omitStyledProps(props)), children);
});
ButtonLayout.displayName = 'ButtonLayout';
export var ButtonItem = styled(ButtonLayout).withConfig({
  displayName: "ButtonItem",
  componentId: "sc-22szay-0"
})(["align-items:center;background:transparent;border:none;color:", ";cursor:pointer;display:inline-flex;font-family:inherit;font-size:", ";height:", ";justify-content:center;margin:0;padding:0 ", ";transition:background ", "ms ease;user-select:none;", " &:active,&:hover{background:", ";color:", ";}&:focus{outline:none;}&[disabled]{color:", ";cursor:default;&:hover{background:inherit;}}&[aria-pressed='true']{background:", ";color:", ";&[disabled]{background:", ";color:", ";}}"], function (_ref2) {
  var theme = _ref2.theme;
  return theme.colors.text3;
}, function (_ref3) {
  var theme = _ref3.theme;
  return theme.fontSizes.small;
}, inputHeight, function (_ref4) {
  var theme = _ref4.theme;
  return theme.space.u3;
}, function (_ref5) {
  var theme = _ref5.theme;
  return theme.transitions.quick;
}, space, function (_ref6) {
  var theme = _ref6.theme;
  return theme.colors.neutralSubtle;
}, function (_ref7) {
  var theme = _ref7.theme;
  return theme.colors.text5;
}, function (_ref8) {
  var theme = _ref8.theme;
  return theme.colors.text1;
}, function (_ref9) {
  var theme = _ref9.theme;
  return theme.colors.keySubtle;
}, function (_ref10) {
  var theme = _ref10.theme;
  return theme.colors.key;
}, function (_ref11) {
  var theme = _ref11.theme;
  return theme.colors.keySubtle;
}, function (_ref12) {
  var theme = _ref12.theme;
  return theme.colors.keyFocus;
});
