// @looker/components/cjs/ProgressCircular/DeterminateProgress.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DeterminateProgress = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _ProgressSvg = require("./ProgressSvg");

var _size = require("./size");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const DeterminateProgress = ({
  size,
  progress = 0
}) => {
  const _progressCircularSVG = (0, _size.progressCircularSVG)({
    size
  }),
        stroke = _progressCircularSVG.stroke,
        half = _progressCircularSVG.half,
        radius = _progressCircularSVG.radius,
        dashArray = _progressCircularSVG.dashArray;

  const progressOffset = (1 - progress) * (2 * Math.PI * radius);
  return _react.default.createElement(DeterminateContainer, null, _react.default.createElement(_ProgressSvg.CircleContainer, {
    viewBox: `0 0 ${half * 2} ${half * 2}`,
    xmlns: "http://www.w3.org/2000/svg"
  }, _react.default.createElement(DeterminateCircle, {
    cx: half,
    cy: half,
    r: radius,
    strokeDasharray: dashArray,
    strokeDashoffset: progressOffset,
    strokeWidth: stroke
  })));
};

exports.DeterminateProgress = DeterminateProgress;

const DeterminateContainer = _styledComponents.default.div.withConfig({
  displayName: "DeterminateProgress__DeterminateContainer",
  componentId: "sc-yb0ja2-0"
})(["height:100%;position:absolute;transform:rotate(-90deg);width:100%;"]);

const DeterminateCircle = _styledComponents.default.circle.withConfig({
  displayName: "DeterminateProgress__DeterminateCircle",
  componentId: "sc-yb0ja2-1"
})(["stroke:", ";transition:stroke-dashoffset 500ms;"], ({
  theme
}) => theme.colors.key);
