// @looker/components/Popover/Layout/PopoverLayout.js
import React from 'react';
import { ModalLoading } from '../../Modal/ModalLayout';
import { PopoverContent } from './PopoverContent';
import { PopoverFooter } from './PopoverFooter';
import { PopoverHeader } from './PopoverHeader';
export var PopoverLayout = function PopoverLayout(_ref) {
  var children = _ref.children,
      closeButton = _ref.closeButton,
      _ref$footer = _ref.footer,
      footer = _ref$footer === void 0 ? true : _ref$footer,
      header = _ref.header,
      _ref$hideHeader = _ref.hideHeader,
      hideHeader = _ref$hideHeader === void 0 ? false : _ref$hideHeader,
      isLoading = _ref.isLoading;
  var internalFooter = typeof footer === 'boolean' ? null : footer;
  return React.createElement(React.Fragment, null, header && React.createElement(PopoverHeader, {
    hidden: hideHeader,
    hideClose: !!footer
  }, header), React.createElement(PopoverContent, {
    hasFooter: !!footer,
    hasHeader: !!header
  }, isLoading ? React.createElement(ModalLoading, null) : children), footer && React.createElement(PopoverFooter, {
    closeButton: closeButton
  }, internalFooter));
};
