// @looker/components/cjs/LkFieldTree/LkFieldItemContent.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.LkFieldItemContent = void 0;

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _ListItem = require("../ListItem");

var _utils = require("../Tree/utils");

var _defaults = require("./defaults");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const LkFieldItemContent = _styledComponents.default.div.attrs(({
  role = 'treeitem'
}) => ({
  role
})).withConfig({
  displayName: "LkFieldItemContent",
  componentId: "sc-vth29c-0"
})(["", "{padding:0;padding-left:", ";}", " ", " background:none;color:inherit;cursor:", ";display:flex;flex:1;font-size:inherit;font-weight:inherit;margin:0;min-width:0;outline:none;position:relative;text-align:left;text-decoration:none;width:100%;&:hover,&:focus{color:inherit;text-decoration:none;}&[disabled]{color:", ";cursor:not-allowed;}"], _ListItem.ListItemLabel, ({
  theme
}) => theme.space.xxsmall, ({
  depth
}) => (0, _utils.generateIndent)({
  density: _defaults.lkFieldItemDensity,
  depth
}), ({
  focusVisible,
  theme
}) => focusVisible && `
    &::after {
      bottom: 0;
      box-shadow: inset 0 0 0 2px ${theme.colors.keyFocus};
      content: '';
      left: 0;
      position: absolute;
      right: 0;
      top: 0;
      z-index: 1;
    }
  `, ({
  cursorPointer
}) => cursorPointer ? 'pointer' : undefined, ({
  theme
}) => theme.colors.text1);

exports.LkFieldItemContent = LkFieldItemContent;
