// @looker/components/cjs/Dialog/Layout/DialogLayout.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DialogLayout = void 0;

var _react = _interopRequireDefault(require("react"));

var _ModalLayout = require("../../Modal/ModalLayout");

var _DialogContent = require("./DialogContent");

var _DialogFooter = require("./DialogFooter");

var _DialogHeader = require("./DialogHeader");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const constructDialogHeader = (children, closeButton, detail, footerExists) => {
  if (!children) return null;
  const props = {
    children
  };

  if (detail) {
    props.detail = detail;
  } else if (closeButton || footerExists) {
    props.hideClose = !closeButton;
  }

  return _react.default.createElement(_DialogHeader.DialogHeader, props);
};

const DialogLayout = ({
  children,
  footer,
  footerSecondary,
  header,
  headerCloseButton = !footer && true,
  headerDetail,
  isLoading
}) => {
  const dialogFooter = footer ? _react.default.createElement(_DialogFooter.DialogFooter, {
    secondary: footerSecondary
  }, footer) : null;
  const dialogHeader = constructDialogHeader(header, headerCloseButton, headerDetail, !!footer);
  return _react.default.createElement(_ModalLayout.ModalLayout, {
    footer: dialogFooter,
    header: dialogHeader
  }, _react.default.createElement(_DialogContent.DialogContent, {
    hasFooter: !!footer,
    hasHeader: !!header
  }, isLoading ? _react.default.createElement(_ModalLayout.ModalLoading, null) : children));
};

exports.DialogLayout = DialogLayout;
