// @looker/components/cjs/UnorderedList/UnorderedList.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.UnorderedList = void 0;

var _designTokens = require("@looker/design-tokens");

var _styledComponents = _interopRequireDefault(require("styled-components"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const typeVariant = (0, _designTokens.variant)({
  prop: 'type',
  variants: {
    bullet: {
      listStyleType: 'disc',
      pl: 'u4'
    },
    none: {
      listStyleType: 'none'
    }
  }
});

const UnorderedList = _styledComponents.default.ul.withConfig({
  shouldForwardProp: _designTokens.shouldForwardProp
}).attrs(({
  color = 'body',
  type = 'none'
}) => ({
  color,
  type
})).withConfig({
  displayName: "UnorderedList",
  componentId: "sc-qnps5j-0"
})(["", " ", " ", " ", " ", " ", " ", " li{margin-bottom:", ";}"], _designTokens.reset, _designTokens.textColor, _designTokens.typography, typeVariant, _designTokens.space, _designTokens.position, _designTokens.layout, ({
  theme
}) => theme.space.u1);

exports.UnorderedList = UnorderedList;
