// @looker/components/Ripple/useRippleHandlers.js
import { useCallback } from 'react';
import { useWrapEvent } from '../utils';
export var rippleHandlerKeys = ['onBlur', 'onFocus', 'onKeyDown', 'onKeyUp', 'onMouseDown', 'onMouseEnter', 'onMouseLeave', 'onMouseUp'];
export var useRippleHandlers = function useRippleHandlers(_ref, currentHandlers, disabled) {
  var startBG = _ref.startBG,
      endBG = _ref.endBG,
      startFG = _ref.startFG,
      endFG = _ref.endFG;
  var handleKeyDown = useCallback(function (e) {
    switch (e.key) {
      case 'Enter':
      case ' ':
        startFG();
        break;
    }
  }, [startFG]);
  var handleMouseLeave = useCallback(function () {
    endBG();
    endFG();
  }, [endFG, endBG]);
  var wrappedCallbacks = {
    onBlur: useWrapEvent(endBG, currentHandlers.onBlur),
    onFocus: useWrapEvent(startBG, currentHandlers.onFocus),
    onKeyDown: useWrapEvent(handleKeyDown, currentHandlers.onKeyDown),
    onKeyUp: useWrapEvent(endFG, currentHandlers.onKeyUp),
    onMouseDown: useWrapEvent(startFG, currentHandlers.onMouseDown),
    onMouseEnter: useWrapEvent(startBG, currentHandlers.onMouseEnter),
    onMouseLeave: useWrapEvent(handleMouseLeave, currentHandlers.onMouseLeave),
    onMouseUp: useWrapEvent(endFG, currentHandlers.onMouseUp)
  };
  return disabled ? {} : wrappedCallbacks;
};
