// @looker/components/Form/Inputs/Radio/Radio.js
import "core-js/modules/es.object.keys.js";
import "core-js/modules/es.array.index-of.js";
import "core-js/modules/es.symbol.js";
import "core-js/modules/es.object.define-property.js";
import "core-js/modules/es.array.filter.js";
import "core-js/modules/es.object.get-own-property-descriptor.js";
import "core-js/modules/es.array.for-each.js";
import "core-js/modules/web.dom-collections.for-each.js";
import "core-js/modules/es.object.get-own-property-descriptors.js";
import "core-js/modules/es.object.define-properties.js";
import "core-js/modules/es.object.assign.js";
var _excluded = ["className", "style", "validationType"];

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import pick from 'lodash/pick';
import React, { forwardRef } from 'react';
import styled from 'styled-components';
import { reset, space } from '@looker/design-tokens';
import { mergeClassNames } from '../../../utils';
import { inputRippleColor, RIPPLE_RATIO, rippleHandlerKeys, rippleStyle, useRipple, useRippleHandlers } from '../../../Ripple';
import { pickInputProps } from '../InputProps';
import { FauxRadio } from './FauxRadio';
export var Radio = styled(forwardRef(function (props, ref) {
  var className = props.className,
      style = props.style,
      validationType = props.validationType,
      restProps = _objectWithoutProperties(props, _excluded);

  var _useRipple = useRipple({
    color: inputRippleColor(props.checked === true, validationType === 'error'),
    size: RIPPLE_RATIO
  }),
      callbacks = _useRipple.callbacks,
      rippleClassName = _useRipple.className,
      rippleStyle = _useRipple.style;

  var rippleHandlers = useRippleHandlers(callbacks, _objectSpread({}, pick(restProps, rippleHandlerKeys)), restProps.disabled);
  return React.createElement("div", _extends({
    className: mergeClassNames([className, rippleClassName]),
    style: _objectSpread(_objectSpread({}, style), rippleStyle)
  }, rippleHandlers), React.createElement("input", _extends({}, pickInputProps(restProps), {
    "aria-invalid": validationType === 'error' ? 'true' : undefined,
    ref: ref,
    type: "radio"
  })), React.createElement(FauxRadio, null));
})).withConfig({
  displayName: "Radio",
  componentId: "sc-1e6vir3-0"
})(["", " ", " ", " align-items:center;display:flex;height:", ";justify-content:center;position:relative;width:", ";input{height:100%;left:0;margin:0;opacity:0;position:absolute;top:0;width:100%;z-index:1;&:checked + ", "{color:", ";&::after{background:currentColor;}}&[aria-invalid='true'] + ", "{color:", ";}&:disabled{cursor:not-allowed;+ ", ",&:not(:checked):hover + ", "{color:", ";}}&:not(:checked):not([aria-invalid='true']):not(:disabled){&:hover,&:focus{+ ", "{color:", ";}}}}"], reset, space, rippleStyle, function (_ref) {
  var space = _ref.theme.space;
  return space.u6;
}, function (_ref2) {
  var space = _ref2.theme.space;
  return space.u6;
}, FauxRadio, function (_ref3) {
  var theme = _ref3.theme;
  return theme.colors.key;
}, FauxRadio, function (_ref4) {
  var theme = _ref4.theme;
  return theme.colors.critical;
}, FauxRadio, FauxRadio, function (_ref5) {
  var theme = _ref5.theme;
  return theme.colors.ui2;
}, FauxRadio, function (_ref6) {
  var theme = _ref6.theme;
  return theme.colors.ui5;
});
