// @looker/design-tokens/legacy/palette.js
export var white = '#FFFFFF';
export var charcoal000 = '#FBFBFC';
export var charcoal100 = '#F5F6F7';
export var charcoal200 = '#DEE1E5';
export var charcoal300 = '#C1C6CC';
export var charcoal400 = '#939BA5';
export var charcoal500 = '#707781';
export var charcoal600 = '#4C535B';
export var charcoal700 = '#343C42';
export var charcoal800 = '#262D33';
export var charcoal900 = '#262D33';
export var purple000 = '#F3F2FF';
export var purple100 = '#E8E5FF';
export var purple200 = '#BFB2FF';
export var purple300 = '#9785F2';
export var purple400 = '#6C43E0';
export var purple500 = '#4F2ABA';
export var purple600 = '#412399';
export var purple700 = '#341C7A';
export var purple800 = '#291661';
export var purple900 = '#1E1047';
export var blue000 = '#f7fcff';
export var blue100 = '#e0f2ff';
export var blue200 = '#bfe3ff';
export var blue300 = '#6fbcf7';
export var blue400 = '#49a9f2';
export var blue500 = '#0087e1';
export var blue600 = '#0059b2';
export var blue700 = '#00418c';
export var blue800 = '#0f2f66';
export var blue900 = '#08284d';
export var green000 = '#f2fff5';
export var green100 = '#e0ffe7';
export var green200 = '#b4fac2';
export var green300 = '#67e591';
export var green400 = '#33cc73';
export var green500 = '#24b25f';
export var green600 = '#0e8c42';
export var green700 = '#0b6b38';
export var green800 = '#08522d';
export var green900 = '#06381f';
export var yellow000 = '#FFF7E8';
export var yellow100 = '#FFEBC4';
export var yellow200 = '#FFDB96';
export var yellow300 = '#FFCA62';
export var yellow400 = '#FFB72B';
export var yellow500 = '#FFA800';
export var yellow600 = '#EF9E00';
export var yellow700 = '#CC8600';
export var yellow800 = '#A56D00';
export var yellow900 = '#724B00';
export var red000 = '#FFF2F4';
export var red100 = '#FFE5E9';
export var red200 = '#FFB8C1';
export var red300 = '#FF667A';
export var red400 = '#ED3B53';
export var red500 = '#CC1F36';
export var red600 = '#B2121F';
export var red700 = '#990F14';
export var red800 = '#730B0F';
export var red900 = '#52080B';
