// @looker/components/Spinner/Spinner.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import "core-js/modules/es.array.map.js";
import "core-js/modules/es.object.assign.js";
import { position, space, reset, shouldForwardProp } from '@looker/design-tokens';
import omit from 'lodash/omit';
import range from 'lodash/range';
import React from 'react';
import styled from 'styled-components';
import { SpinnerMarker } from './SpinnerMarker';

var SpinnerFactory = function SpinnerFactory(props) {
  var _props$color = props.color,
      color = _props$color === void 0 ? 'text5' : _props$color,
      _props$markers = props.markers,
      markers = _props$markers === void 0 ? 13 : _props$markers,
      markerRadius = props.markerRadius,
      _props$speed = props.speed,
      speed = _props$speed === void 0 ? 1000 : _props$speed;
  return React.createElement(Style, _extends({
    "data-testid": "loading-spinner"
  }, omit(props, 'color', 'markers', 'markersRadius', 'speed')), range(markers).map(function (i) {
    return React.createElement(SpinnerMarker, {
      backgroundColor: color,
      key: i,
      speed: speed,
      markers: markers,
      markerIndex: i,
      markerRadius: markerRadius
    });
  }));
};

var Style = styled.div.withConfig({
  shouldForwardProp: shouldForwardProp
}).attrs(function (_ref) {
  var _ref$size = _ref.size,
      size = _ref$size === void 0 ? '30' : _ref$size;
  return {
    size: size
  };
}).withConfig({
  displayName: "Spinner__Style",
  componentId: "sc-dvoyit-0"
})(["", " ", " ", " height:", "px;position:relative;width:", "px;"], reset, space, position, function (_ref2) {
  var size = _ref2.size;
  return size;
}, function (_ref3) {
  var size = _ref3.size;
  return size;
});
export var Spinner = styled(SpinnerFactory).withConfig({
  displayName: "Spinner",
  componentId: "sc-dvoyit-1"
})([""]);
