// @looker/components/Form/Inputs/OptionsGroup/RadioGroup.js
import "core-js/modules/es.object.keys.js";
import "core-js/modules/es.array.index-of.js";
import "core-js/modules/es.symbol.js";
import "core-js/modules/es.object.assign.js";
var _excluded = ["autoFocus", "defaultValue", "disabled", "inline", "name", "onChange", "options", "validationType", "value"];

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import "core-js/modules/es.function.name.js";
import "core-js/modules/es.array.map.js";
import "core-js/modules/es.object.define-property.js";

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import React, { forwardRef, useCallback } from 'react';
import styled from 'styled-components';
import { useID } from '../../../utils';
import { Fieldset } from '../../Fieldset';
import { FieldRadio } from '../../Fields/FieldRadio';
import { inputHeight } from '../height';

function getCheckedProps(optionValue, isControlled, value, defaultValue) {
  var key = isControlled ? 'checked' : 'defaultChecked';
  var valueToUse = isControlled ? value : defaultValue;
  return _defineProperty({}, key, valueToUse === optionValue);
}

var RadioGroupLayout = forwardRef(function (_ref2, ref) {
  var autoFocus = _ref2.autoFocus,
      defaultValue = _ref2.defaultValue,
      disabled = _ref2.disabled,
      inline = _ref2.inline,
      propsName = _ref2.name,
      onChange = _ref2.onChange,
      options = _ref2.options,
      validationType = _ref2.validationType,
      value = _ref2.value,
      rest = _objectWithoutProperties(_ref2, _excluded);

  var name = useID(propsName);
  var isControlled = onChange !== undefined && defaultValue === undefined;
  var getChangeHandler = useCallback(function (optionValue) {
    return onChange ? function () {
      return onChange(optionValue);
    } : undefined;
  }, [onChange]);
  var radios = options.map(function (option, index) {
    var checkedProps = getCheckedProps(option.value, isControlled, value, defaultValue);
    var autoFocusProps = index === 0 && autoFocus ? {
      autoFocus: autoFocus
    } : {};
    return React.createElement(FieldRadio, _extends({
      detail: option.detail,
      disabled: option.disabled || disabled,
      key: option.value,
      label: option.label,
      name: name,
      onChange: getChangeHandler(option.value),
      validationType: validationType
    }, checkedProps, autoFocusProps));
  });
  return React.createElement(Fieldset, _extends({
    "data-testid": "radio-list",
    inline: inline,
    wrap: inline,
    gap: !inline ? 'u1' : undefined,
    width: inline ? 'auto' : undefined,
    ref: ref
  }, rest), radios);
});
RadioGroupLayout.displayName = 'RadioGroupLayout';
export var RadioGroup = styled(RadioGroupLayout).withConfig({
  displayName: "RadioGroup",
  componentId: "sc-x72kee-0"
})(["", "{", "}"], FieldRadio, function (_ref3) {
  var inline = _ref3.inline;
  return inline ? "line-height: ".concat(inputHeight, ";") : '';
});
