// @looker/components/cjs/Spinner/Spinner.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Spinner = void 0;

require("core-js/modules/es.array.map.js");

var _designTokens = require("@looker/design-tokens");

var _omit = _interopRequireDefault(require("lodash/omit"));

var _range = _interopRequireDefault(require("lodash/range"));

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _SpinnerMarker = require("./SpinnerMarker");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

const SpinnerFactory = props => {
  const _props$color = props.color,
        color = _props$color === void 0 ? 'text5' : _props$color,
        _props$markers = props.markers,
        markers = _props$markers === void 0 ? 13 : _props$markers,
        markerRadius = props.markerRadius,
        _props$speed = props.speed,
        speed = _props$speed === void 0 ? 1000 : _props$speed;
  return _react.default.createElement(Style, _extends({
    "data-testid": "loading-spinner"
  }, (0, _omit.default)(props, 'color', 'markers', 'markersRadius', 'speed')), (0, _range.default)(markers).map(i => _react.default.createElement(_SpinnerMarker.SpinnerMarker, {
    backgroundColor: color,
    key: i,
    speed: speed,
    markers: markers,
    markerIndex: i,
    markerRadius: markerRadius
  })));
};

const Style = _styledComponents.default.div.withConfig({
  shouldForwardProp: _designTokens.shouldForwardProp
}).attrs(({
  size = '30'
}) => ({
  size
})).withConfig({
  displayName: "Spinner__Style",
  componentId: "sc-dvoyit-0"
})(["", " ", " ", " height:", "px;position:relative;width:", "px;"], _designTokens.reset, _designTokens.space, _designTokens.position, ({
  size
}) => size, ({
  size
}) => size);

const Spinner = (0, _styledComponents.default)(SpinnerFactory).withConfig({
  displayName: "Spinner",
  componentId: "sc-dvoyit-1"
})([""]);
exports.Spinner = Spinner;
