// @looker/components/Form/Inputs/InputChips/InputChips.js
var _excluded = ["values", "onChange", "chipIconLabel", "clearIconLabel", "inputValue", "onInputChange", "parseInputValue", "validate", "formatInputValue", "onValidationFail", "onDuplicate", "onBlur", "onKeyDown", "onPaste"];

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _toConsumableArray(arr) { return _arrayWithoutHoles(arr) || _iterableToArray(arr) || _unsupportedIterableToArray(arr) || _nonIterableSpread(); }

function _nonIterableSpread() { throw new TypeError("Invalid attempt to spread non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _iterableToArray(iter) { if (typeof Symbol !== "undefined" && iter[Symbol.iterator] != null || iter["@@iterator"] != null) return Array.from(iter); }

function _arrayWithoutHoles(arr) { if (Array.isArray(arr)) return _arrayLikeToArray(arr); }

function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _iterableToArrayLimit(arr, i) { var _i = arr == null ? null : typeof Symbol !== "undefined" && arr[Symbol.iterator] || arr["@@iterator"]; if (_i == null) return; var _arr = []; var _n = true; var _d = false; var _s, _e; try { for (_i = _i.call(arr); !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import "core-js/modules/es.regexp.exec.js";
import "core-js/modules/es.string.replace.js";
import "core-js/modules/es.array.map.js";
import "core-js/modules/es.string.split.js";
import "core-js/modules/es.regexp.constructor.js";
import "core-js/modules/es.regexp.to-string.js";
import "core-js/modules/es.array.for-each.js";
import "core-js/modules/web.dom-collections.for-each.js";
import "core-js/modules/es.array.includes.js";
import "core-js/modules/es.string.includes.js";
import "core-js/modules/es.string.trim.js";
import "core-js/modules/es.array.iterator.js";
import "core-js/modules/es.object.to-string.js";
import "core-js/modules/web.dom-collections.iterator.js";
import "core-js/modules/es.array.join.js";
import "core-js/modules/es.array.concat.js";
import "core-js/modules/es.string.ends-with.js";
import "core-js/modules/es.object.keys.js";
import "core-js/modules/es.array.index-of.js";
import "core-js/modules/es.symbol.js";
import "core-js/modules/es.array.is-array.js";
import "core-js/modules/es.symbol.description.js";
import "core-js/modules/es.symbol.iterator.js";
import "core-js/modules/es.string.iterator.js";
import "core-js/modules/es.array.slice.js";
import "core-js/modules/es.function.name.js";
import "core-js/modules/es.array.from.js";
import "core-js/modules/es.object.assign.js";
import React, { forwardRef, useRef, useState } from 'react';
import styled from 'styled-components';
import { useControlWarn, useWrapEvent } from '../../../utils';
import { InputChipsBase } from './InputChipsBase';
export var splitInputValue = function splitInputValue(inputValue) {
  var commaKey = '0UX1bJKsFBFQonIIXq9oyeV40ITHwtew';
  var tabKey = 'heF6X4qMBtIti8c8U9nMhskYOQUQiXqx';
  var removedEscapes = inputValue.replace(/\\,/g, commaKey).replace(/\\\t/g, tabKey);
  var splitRegexp = "[,\\t\\n\\r]+";
  return removedEscapes.split(new RegExp(splitRegexp)).map(function (value) {
    return value.replace(new RegExp(commaKey, 'g'), ',').replace(new RegExp(tabKey, 'g'), '\t');
  });
};
export var validateValues = function validateValues(newValues, currentValues, validate, formatInputValue) {
  var duplicateValues = [];
  var invalidValues = [];
  var unusedValues = [];
  var validValues = [];
  newValues.forEach(function (val) {
    var formattedValue = formatInputValue ? formatInputValue(val) : val;
    if (formattedValue === '') return;

    if (validate && !validate(formattedValue)) {
      unusedValues.push(formattedValue);
      return invalidValues.push(formattedValue);
    } else if (currentValues && currentValues.includes(formattedValue)) {
      unusedValues.push(formattedValue);
      return duplicateValues.push(formattedValue);
    } else {
      return validValues.push(formattedValue);
    }
  });
  return {
    duplicateValues: duplicateValues,
    invalidValues: invalidValues,
    unusedValues: unusedValues,
    validValues: validValues
  };
};

var trimValue = function trimValue(value) {
  return value.trim();
};

export var InputChips = styled(forwardRef(function (_ref, ref) {
  var values = _ref.values,
      onChange = _ref.onChange,
      chipIconLabel = _ref.chipIconLabel,
      clearIconLabel = _ref.clearIconLabel,
      controlledInputValue = _ref.inputValue,
      onInputChange = _ref.onInputChange,
      _ref$parseInputValue = _ref.parseInputValue,
      parseInputValue = _ref$parseInputValue === void 0 ? splitInputValue : _ref$parseInputValue,
      validate = _ref.validate,
      _ref$formatInputValue = _ref.formatInputValue,
      formatInputValue = _ref$formatInputValue === void 0 ? trimValue : _ref$formatInputValue,
      onValidationFail = _ref.onValidationFail,
      onDuplicate = _ref.onDuplicate,
      onBlur = _ref.onBlur,
      onKeyDown = _ref.onKeyDown,
      onPaste = _ref.onPaste,
      props = _objectWithoutProperties(_ref, _excluded);

  var isControlled = useControlWarn({
    controllingProps: ['inputValue', 'onInputChange'],
    isControlledCheck: function isControlledCheck() {
      return controlledInputValue !== undefined && onInputChange !== undefined;
    },
    name: 'InputChips'
  });

  var _useState = useState(''),
      _useState2 = _slicedToArray(_useState, 2),
      uncontrolledValue = _useState2[0],
      setUncontrolledValue = _useState2[1];

  var inputValue = isControlled ? controlledInputValue || '' : uncontrolledValue;

  var setInputValue = function setInputValue(val, event) {
    if (!isControlled) {
      setUncontrolledValue(val);
    }

    if (val !== inputValue) {
      onInputChange && onInputChange(val, event);
    }
  };

  var updateValues = function updateValues(newInputValue) {
    var inputValues = parseInputValue(newInputValue || inputValue);

    var _validateValues = validateValues(inputValues, values, validate, formatInputValue),
        duplicateValues = _validateValues.duplicateValues,
        invalidValues = _validateValues.invalidValues,
        unusedValues = _validateValues.unusedValues,
        validValues = _validateValues.validValues;

    var updatedInputValue = unusedValues.join(', ');
    var updatedValues = validValues.length && [].concat(_toConsumableArray(values), _toConsumableArray(validValues));

    if (updatedValues) {
      onChange(updatedValues);
    }

    if (invalidValues.length > 0) {
      onValidationFail && onValidationFail(invalidValues);
    }

    if (duplicateValues.length > 0) {
      onDuplicate && onDuplicate(duplicateValues);
    }

    setInputValue(updatedInputValue);
  };

  var handleBlur = function handleBlur() {
    updateValues();
  };

  var handleKeyDown = function handleKeyDown(e) {
    if (e.key === 'Enter') {
      e.preventDefault();
      updateValues();
    }
  };

  var pastedValue = useRef();

  var handlePaste = function handlePaste(e) {
    pastedValue.current = e.clipboardData.getData('Text');
  };

  var handleInputChange = function handleInputChange(value, event) {
    if (pastedValue.current || value.endsWith(',') && !value.endsWith('\\,')) {
      updateValues(pastedValue.current || value);
      pastedValue.current = null;
    } else {
      setInputValue(value, event);
    }
  };

  var wrappedEvents = {
    onBlur: useWrapEvent(handleBlur, onBlur),
    onKeyDown: useWrapEvent(handleKeyDown, onKeyDown),
    onPaste: useWrapEvent(handlePaste, onPaste)
  };
  return React.createElement(InputChipsBase, _extends({
    ref: ref,
    values: values,
    onChange: onChange,
    chipIconLabel: chipIconLabel,
    clearIconLabel: clearIconLabel,
    inputValue: inputValue,
    onInputChange: handleInputChange
  }, wrappedEvents, props));
})).withConfig({
  displayName: "InputChips",
  componentId: "sc-6zpztz-0"
})([""]);
