// @looker/components/cjs/DataTable/Item/DataTableCheckbox.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DataTableCheckbox = exports.checkListProps = void 0;

var _react = _interopRequireDefault(require("react"));

var _Form = require("../../Form");

var _ItemTarget = require("./ItemTarget");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const checkListProps = ['checked', 'disabled', 'onChange', 'id'];
exports.checkListProps = checkListProps;

const DataTableCheckbox = ({
  id,
  onChange,
  checked,
  disabled
}) => {
  const handleCellOnClick = () => !disabled && onChange && onChange();

  const handleOnKeyDown = event => event.key === 'Enter' && event.currentTarget.click();

  let ariaLabel = 'Select all rows';

  if (id !== 'headerId') {
    ariaLabel = undefined;
  } else if (checked) {
    ariaLabel = 'Select none';
  }

  return _react.default.createElement(_ItemTarget.ItemTarget, {
    "aria-labelledby": `rowheader-${id}`,
    onClick: handleCellOnClick
  }, _react.default.createElement(_Form.Checkbox, {
    "aria-label": ariaLabel,
    checked: checked,
    disabled: disabled,
    onKeyDown: handleOnKeyDown,
    tabIndex: -1
  }));
};

exports.DataTableCheckbox = DataTableCheckbox;
