// @looker/components/cjs/ListItem/utils/listItemDimensions.js
"use strict";

require("core-js/modules/es.array.filter.js");

require("core-js/modules/es.object.get-own-property-descriptors.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.listItemDimensions = exports.densities = exports.densityNegative3 = exports.densityNegative2 = exports.densityNegative1 = exports.densityPositive1 = exports.density0 = void 0;

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

const density0 = {
  descriptionFontSize: 'xsmall',
  descriptionLineHeight: 'xsmall',
  height: 36,
  iconGap: 'u3',
  iconSize: 'small',
  labelFontSize: 'small',
  labelLineHeight: 'small',
  px: 'u4',
  py: 'u2'
};
exports.density0 = density0;

const densityPositive1 = _objectSpread(_objectSpread({}, density0), {}, {
  descriptionFontSize: 'small',
  descriptionLineHeight: 'small',
  height: 48,
  iconGap: 'u4',
  iconSize: 'medium',
  labelFontSize: 'medium',
  labelLineHeight: 'medium',
  py: 'u3'
});

exports.densityPositive1 = densityPositive1;

const densityNegative1 = _objectSpread(_objectSpread({}, density0), {}, {
  height: 32,
  py: '0.375rem'
});

exports.densityNegative1 = densityNegative1;

const densityNegative2 = _objectSpread(_objectSpread({}, densityNegative1), {}, {
  height: 28,
  py: 'u1'
});

exports.densityNegative2 = densityNegative2;

const densityNegative3 = _objectSpread(_objectSpread({}, densityNegative2), {}, {
  height: 24,
  iconGap: 'u2',
  iconSize: 'xxsmall',
  labelFontSize: 'xsmall',
  labelLineHeight: 'xsmall'
});

exports.densityNegative3 = densityNegative3;
const densities = {
  '-1': densityNegative1,
  '-2': densityNegative2,
  '-3': densityNegative3,
  '0': density0,
  '1': densityPositive1
};
exports.densities = densities;

const listItemDimensions = density => densities[density];

exports.listItemDimensions = listItemDimensions;
