// @looker/components/DataTable/Item/DataTableCheckbox.js
import React from 'react';
import { Checkbox } from '../../Form';
import { ItemTarget } from './ItemTarget';
export var checkListProps = ['checked', 'disabled', 'onChange', 'id'];
export var DataTableCheckbox = function DataTableCheckbox(_ref) {
  var id = _ref.id,
      onChange = _ref.onChange,
      checked = _ref.checked,
      disabled = _ref.disabled;

  var handleCellOnClick = function handleCellOnClick() {
    return !disabled && onChange && onChange();
  };

  var handleOnKeyDown = function handleOnKeyDown(event) {
    return event.key === 'Enter' && event.currentTarget.click();
  };

  var ariaLabel = 'Select all rows';

  if (id !== 'headerId') {
    ariaLabel = undefined;
  } else if (checked) {
    ariaLabel = 'Select none';
  }

  return React.createElement(ItemTarget, {
    "aria-labelledby": "rowheader-".concat(id),
    onClick: handleCellOnClick
  }, React.createElement(Checkbox, {
    "aria-label": ariaLabel,
    checked: checked,
    disabled: disabled,
    onKeyDown: handleOnKeyDown,
    tabIndex: -1
  }));
};
