// @looker/components/Form/Fields/FieldText/FieldText.js
import "core-js/modules/es.array.is-array.js";
import "core-js/modules/es.symbol.js";
import "core-js/modules/es.symbol.description.js";
import "core-js/modules/es.object.to-string.js";
import "core-js/modules/es.symbol.iterator.js";
import "core-js/modules/es.array.iterator.js";
import "core-js/modules/es.string.iterator.js";
import "core-js/modules/web.dom-collections.iterator.js";
import "core-js/modules/es.array.slice.js";
import "core-js/modules/es.function.name.js";
import "core-js/modules/es.array.from.js";
import "core-js/modules/es.object.assign.js";

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _iterableToArrayLimit(arr, i) { var _i = arr == null ? null : typeof Symbol !== "undefined" && arr[Symbol.iterator] || arr["@@iterator"]; if (_i == null) return; var _arr = []; var _n = true; var _d = false; var _s, _e; try { for (_i = _i.call(arr); !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

import React, { forwardRef, useContext, useState } from 'react';
import styled, { ThemeContext } from 'styled-components';
import { useID } from '../../../utils';
import { useFormContext } from '../../Form';
import { InputText, InputTextContext } from '../../Inputs/InputText';
import { FloatingLabelField, omitFieldProps, pickFieldProps } from '../Field';
import { getHasValue } from '../Field/useFloatingLabel';
var FieldTextComponent = forwardRef(function (props, ref) {
  var id = useID(props.id);
  var validationMessage = useFormContext(props);

  var _useContext = useContext(ThemeContext),
      space = _useContext.space;

  var _useState = useState(0),
      _useState2 = _slicedToArray(_useState, 2),
      beforeWidth = _useState2[0],
      setBeforeWidth = _useState2[1];

  var labelOffset;

  if (props.iconBefore) {
    labelOffset = space.u8;
  } else if (props.before) {
    labelOffset = "".concat(beforeWidth, "px");
  }

  return React.createElement(InputTextContext.Provider, {
    value: {
      beforeWidth: beforeWidth,
      setBeforeWidth: setBeforeWidth
    }
  }, React.createElement(FloatingLabelField, _extends({
    id: id,
    validationMessage: validationMessage,
    hasValue: getHasValue(props),
    labelOffset: labelOffset
  }, pickFieldProps(props)), React.createElement(InputText, _extends({}, omitFieldProps(props), {
    id: id,
    "aria-describedby": "describedby-".concat(id),
    validationType: validationMessage && validationMessage.type,
    ref: ref
  }))));
});
FieldTextComponent.displayName = 'FieldTextComponent';
export var FieldText = styled(FieldTextComponent).withConfig({
  displayName: "FieldText",
  componentId: "sc-rg64oz-0"
})([""]);
