// @looker/components/Accordion/AccordionDisclosure.js
import "core-js/modules/es.object.assign.js";

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React from 'react';
import styled from 'styled-components';
import { typography } from '@looker/design-tokens';
import { simpleLayoutCSS } from '../Layout/utils/simple';
import { Accordion2Disclosure } from '../Accordion2/Accordion2Disclosure';

var AccordionDisclosureInternal = function AccordionDisclosureInternal(props) {
  return React.createElement(Accordion2Disclosure, _extends({
    focusVisible: false,
    indicator: null
  }, props));
};

export var AccordionDisclosure = styled(AccordionDisclosureInternal).attrs(function (_ref) {
  var _ref$px = _ref.px,
      px = _ref$px === void 0 ? 'none' : _ref$px,
      _ref$py = _ref.py,
      py = _ref$py === void 0 ? 'xsmall' : _ref$py;
  return {
    px: px,
    py: py
  };
}).withConfig({
  displayName: "AccordionDisclosure",
  componentId: "sc-8407zi-0"
})(["", " ", ""], typography, simpleLayoutCSS);
