// @looker/components/Panel/PanelHeader.js
import React, { useContext } from 'react';
import { useTranslation } from 'react-i18next';
import { ThemeContext } from 'styled-components';
import { ArrowBack } from '@styled-icons/material-rounded/ArrowBack';
import { Heading } from '../Text';
import { Space } from '../Layout';
import { IconButton } from '../Button';
export var PanelHeader = function PanelHeader(_ref) {
  var closeLabel = _ref.closeLabel,
      onClose = _ref.onClose,
      _ref$iconToggle = _ref.iconToggle,
      iconToggle = _ref$iconToggle === void 0 ? false : _ref$iconToggle,
      title = _ref.title;

  var _useTranslation = useTranslation('PanelHeader'),
      t = _useTranslation.t;

  var defaultLabel = t('CloseTitle', {
    title: title
  });

  var _useContext = useContext(ThemeContext),
      space = _useContext.space;

  return React.createElement(Space, {
    as: "header",
    height: space.u10,
    px: "large",
    gap: "u3",
    mt: "small",
    mb: "1.5rem",
    flexShrink: 0
  }, React.createElement(IconButton, {
    icon: React.createElement(ArrowBack, null),
    label: closeLabel || defaultLabel,
    onClick: onClose,
    toggle: iconToggle,
    toggleBackground: iconToggle,
    shape: "round",
    size: "small"
  }), React.createElement(Heading, {
    fontSize: "xlarge"
  }, title));
};
