// @looker/components/cjs/utils/useArrowKeyNav.js
"use strict";

require("core-js/modules/es.symbol.description.js");

require("core-js/modules/es.array.iterator.js");

require("core-js/modules/es.array.slice.js");

require("core-js/modules/es.array.from.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useArrowKeyNav = void 0;

var _react = require("react");

var _checkElementRemoved = require("./checkElementRemoved");

var _getNextFocus = require("./getNextFocus");

var _useForkedRef = require("./useForkedRef");

var _useWrapEvent = require("./useWrapEvent");

function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }

function _iterableToArrayLimit(arr, i) { var _i = arr == null ? null : typeof Symbol !== "undefined" && arr[Symbol.iterator] || arr["@@iterator"]; if (_i == null) return; var _arr = []; var _n = true; var _d = false; var _s, _e; try { for (_i = _i.call(arr); !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

const useArrowKeyNav = ({
  axis = 'vertical',
  disabled,
  getNextFocus = _getNextFocus.getNextFocus,
  ref,
  onBlur,
  onFocus,
  onKeyDown
}) => {
  const internalRef = (0, _react.useRef)(null);
  const focusedItemRef = (0, _react.useRef)();

  const _useState = (0, _react.useState)(false),
        _useState2 = _slicedToArray(_useState, 2),
        focusInside = _useState2[0],
        setFocusInside = _useState2[1];

  const handleArrowKey = (e, direction, vertical) => {
    if (internalRef.current) {
      const newFocusedItem = getNextFocus(direction, internalRef.current, vertical);

      if (newFocusedItem) {
        e.preventDefault();
        newFocusedItem.focus();
      }
    }
  };

  const handleKeyDown = e => {
    switch (e.key) {
      case 'ArrowUp':
        axis !== 'horizontal' && handleArrowKey(e, -1, true);
        break;

      case 'ArrowDown':
        axis !== 'horizontal' && handleArrowKey(e, 1, true);
        break;

      case 'ArrowLeft':
        axis !== 'vertical' && handleArrowKey(e, -1, false);
        break;

      case 'ArrowRight':
        axis !== 'vertical' && handleArrowKey(e, 1, false);
        break;
    }
  };

  const placeInitialFocus = (0, _react.useCallback)(() => {
    if (internalRef.current) {
      const toFocus = getNextFocus(1, internalRef.current);

      if (toFocus) {
        toFocus.focus({
          preventScroll: true
        });
      }
    }
  }, [getNextFocus]);

  const handleFocus = e => {
    setFocusInside(true);

    if (e.target === internalRef.current) {
      if (focusedItemRef.current && internalRef.current.contains(focusedItemRef.current)) {
        focusedItemRef.current.focus();
      } else {
        placeInitialFocus();
      }
    } else {
      focusedItemRef.current = e.target;
    }
  };

  const handleBlur = () => {
    setFocusInside(false);
  };

  (0, _react.useEffect)(() => {
    const element = internalRef.current;
    const observer = new MutationObserver(mutationsList => {
      if ((0, _checkElementRemoved.checkElementRemoved)(mutationsList, focusedItemRef.current)) {
        placeInitialFocus();
      }
    });

    if (focusInside && element) {
      observer.observe(element, {
        childList: true,
        subtree: true
      });
    }

    return () => {
      observer.disconnect();
    };
  }, [focusInside, placeInitialFocus]);
  const navProps = {
    onBlur: (0, _useWrapEvent.useWrapEvent)(handleBlur, onBlur),
    onFocus: (0, _useWrapEvent.useWrapEvent)(handleFocus, onFocus),
    onKeyDown: (0, _useWrapEvent.useWrapEvent)(handleKeyDown, onKeyDown),
    ref: (0, _useForkedRef.useForkedRef)(internalRef, ref),
    tabIndex: focusInside ? undefined : 0
  };
  return disabled ? {} : navProps;
};

exports.useArrowKeyNav = useArrowKeyNav;
