// @looker/components/utils/useMouseDragPosition.js
import "core-js/modules/es.array.is-array.js";
import "core-js/modules/es.symbol.js";
import "core-js/modules/es.symbol.description.js";
import "core-js/modules/es.object.to-string.js";
import "core-js/modules/es.symbol.iterator.js";
import "core-js/modules/es.array.iterator.js";
import "core-js/modules/es.string.iterator.js";
import "core-js/modules/web.dom-collections.iterator.js";
import "core-js/modules/es.array.slice.js";
import "core-js/modules/es.function.name.js";
import "core-js/modules/es.array.from.js";

function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _iterableToArrayLimit(arr, i) { var _i = arr == null ? null : typeof Symbol !== "undefined" && arr[Symbol.iterator] || arr["@@iterator"]; if (_i == null) return; var _arr = []; var _n = true; var _d = false; var _s, _e; try { for (_i = _i.call(arr); !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

import { useState, useEffect } from 'react';
import throttle from 'lodash/throttle';
import get from 'lodash/get';
export function useMouseDragPosition(targetRef) {
  var _useState = useState(false),
      _useState2 = _slicedToArray(_useState, 2),
      isMouseDown = _useState2[0],
      setIsMouseDown = _useState2[1];

  var _useState3 = useState({
    x: 0,
    y: 0
  }),
      _useState4 = _slicedToArray(_useState3, 2),
      mousePos = _useState4[0],
      setMousePos = _useState4[1];

  var updateMousePos = function updateMousePos(e) {
    var event = get(e, 'touches[0]', e);
    var pageX = event.pageX,
        clientX = event.clientX,
        pageY = event.pageY,
        clientY = event.clientY;
    setMousePos({
      x: pageX || clientX,
      y: pageY || clientY
    });
  };

  var handleStart = function handleStart(e) {
    requestAnimationFrame(function () {
      setIsMouseDown(true);
    });
    updateMousePos(e);
  };

  var handleMove = throttle(updateMousePos, 100);

  var handleEnd = function handleEnd() {
    requestAnimationFrame(function () {
      setIsMouseDown(false);
    });
  };

  useEffect(function () {
    targetRef && targetRef.addEventListener('mousedown', handleStart);
    targetRef && targetRef.addEventListener('touchstart', handleStart);
    window.addEventListener('mouseup', handleEnd);
    window.addEventListener('touchend', handleEnd);

    if (isMouseDown) {
      window.addEventListener('touchmove', handleMove);
      window.addEventListener('mousemove', handleMove);
      window.addEventListener('mouseleave', handleEnd);
    }

    return function () {
      targetRef && targetRef.removeEventListener('mousedown', handleStart);
      targetRef && targetRef.removeEventListener('touchstart', handleStart);
      window.removeEventListener('mouseup', handleEnd);
      window.removeEventListener('touchend', handleEnd);

      if (isMouseDown) {
        window.removeEventListener('touchmove', handleMove);
        window.removeEventListener('mousemove', handleMove);
        window.removeEventListener('mouseleave', handleEnd);
      }
    };
  }, [isMouseDown, targetRef]);
  return {
    isMouseDown: isMouseDown,
    mousePos: mousePos
  };
}
