// @looker/components/Form/Inputs/InlineTextArea/InlineTextArea.js
import "core-js/modules/es.object.keys.js";
import "core-js/modules/es.array.index-of.js";
import "core-js/modules/es.symbol.js";
import "core-js/modules/es.array.is-array.js";
import "core-js/modules/es.symbol.description.js";
import "core-js/modules/es.object.to-string.js";
import "core-js/modules/es.symbol.iterator.js";
import "core-js/modules/es.array.iterator.js";
import "core-js/modules/es.string.iterator.js";
import "core-js/modules/web.dom-collections.iterator.js";
import "core-js/modules/es.array.slice.js";
import "core-js/modules/es.function.name.js";
import "core-js/modules/es.array.from.js";
import "core-js/modules/es.object.assign.js";
var _excluded = ["className", "onChange", "underlineOnlyOnHover", "value", "placeholder"];

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _iterableToArrayLimit(arr, i) { var _i = arr == null ? null : typeof Symbol !== "undefined" && arr[Symbol.iterator] || arr["@@iterator"]; if (_i == null) return; var _arr = []; var _n = true; var _d = false; var _s, _e; try { for (_i = _i.call(arr); !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React, { forwardRef, useState } from 'react';
import isFunction from 'lodash/isFunction';
import styled from 'styled-components';
import { typography } from '@looker/design-tokens';
import { pickInputProps } from '../InputProps';
export var InlineTextAreaLayout = forwardRef(function (_ref, ref) {
  var className = _ref.className,
      onChange = _ref.onChange,
      underlineOnlyOnHover = _ref.underlineOnlyOnHover,
      _ref$value = _ref.value,
      valueProp = _ref$value === void 0 ? '' : _ref$value,
      placeholder = _ref.placeholder,
      props = _objectWithoutProperties(_ref, _excluded);

  var _useState = useState(valueProp),
      _useState2 = _slicedToArray(_useState, 2),
      value = _useState2[0],
      setValueChange = _useState2[1];

  var displayValue = isFunction(onChange) ? valueProp : value;

  var handleValueChange = function handleValueChange(event) {
    setValueChange(event.currentTarget.value);
  };

  var handleChange = isFunction(onChange) ? onChange : handleValueChange;
  return React.createElement("div", {
    className: className,
    "data-testid": "inline-text-area"
  }, React.createElement(Input, _extends({
    onChange: handleChange,
    ref: ref,
    underlineOnlyOnHover: underlineOnlyOnHover,
    value: displayValue
  }, pickInputProps(props))), React.createElement(VisibleText, {
    displayValue: displayValue
  }, displayValue || placeholder));
});
InlineTextAreaLayout.displayName = 'InlineTextAreaLayout';
var Input = styled.textarea.withConfig({
  displayName: "InlineTextArea__Input",
  componentId: "sc-1ioqw6m-0"
})(["background:transparent;border:none;caret-color:", ";color:transparent;cursor:", ";font:inherit;height:100%;left:0;margin:0;outline:none;padding:0;position:absolute;resize:none;text-align:inherit;text-transform:inherit;top:0;vertical-align:top;width:100%;"], function (_ref2) {
  var theme = _ref2.theme;
  return theme.colors.text5;
}, function (_ref3) {
  var readOnly = _ref3.readOnly,
      disabled = _ref3.disabled;
  return readOnly || disabled ? 'not-allowed' : undefined;
});
var VisibleText = styled.div.withConfig({
  displayName: "InlineTextArea__VisibleText",
  componentId: "sc-1ioqw6m-1"
})(["color:", ";"], function (_ref4) {
  var displayValue = _ref4.displayValue,
      theme = _ref4.theme;
  return displayValue ? 'inherit' : theme.colors.text1;
});
export var InlineTextArea = styled(InlineTextAreaLayout).withConfig({
  displayName: "InlineTextArea",
  componentId: "sc-1ioqw6m-2"
})(["", " border:none;border-bottom:1px dashed;border-bottom-color:", ";color:", ";display:inline-flex;flex-direction:column;justify-content:center;min-height:", ";min-width:2rem;position:relative;text-align:inherit;white-space:pre-wrap;:focus,:hover{background-color:", ";border-bottom-color:", ";outline:none;}:focus{border-bottom-style:solid;}:disabled,:hover{border-bottom-color:", ";}:hover{border-bottom-color:", ";}"], typography, function (_ref5) {
  var theme = _ref5.theme,
      underlineOnlyOnHover = _ref5.underlineOnlyOnHover,
      readOnly = _ref5.readOnly;
  return underlineOnlyOnHover || readOnly ? 'transparent' : theme.colors.ui3;
}, function (_ref6) {
  var disabled = _ref6.disabled,
      theme = _ref6.theme;
  return disabled ? theme.colors.text1 : 'inherit';
}, function (_ref7) {
  var theme = _ref7.theme;
  return theme.lineHeights.medium;
}, function (_ref8) {
  var theme = _ref8.theme;
  return theme.colors.ui1;
}, function (_ref9) {
  var theme = _ref9.theme;
  return theme.colors.key;
}, function (_ref10) {
  var theme = _ref10.theme;
  return theme.colors.text1;
}, function (_ref11) {
  var readOnly = _ref11.readOnly;
  return readOnly && 'transparent';
});
