// @looker/components/Form/Inputs/InputFilters/InputFiltersChip.js
var _excluded = ["filter", "onDelete"];

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import "core-js/modules/es.regexp.exec.js";
import "core-js/modules/es.string.replace.js";
import "core-js/modules/es.array.filter.js";
import "core-js/modules/es.object.keys.js";
import "core-js/modules/es.array.index-of.js";
import "core-js/modules/es.symbol.js";
import "core-js/modules/es.object.assign.js";
import React, { forwardRef } from 'react';
import { Chip } from '../../../Chip';

var defaultFormatValue = function defaultFormatValue(value) {
  return value ? value.replace(/(,(?=\S)|:)/, ', ') : undefined;
};

export var InputFiltersChip = forwardRef(function (_ref, ref) {
  var filter = _ref.filter,
      onDelete = _ref.onDelete,
      props = _objectWithoutProperties(_ref, _excluded);

  var handleDelete = function handleDelete() {
    return onDelete(filter);
  };

  return React.createElement(Chip, _extends({
    onDelete: handleDelete,
    prefix: filter.field,
    ref: ref
  }, props), filter.formatValue && filter.value ? filter.formatValue(filter.value) : defaultFormatValue(filter.value));
});
InputFiltersChip.displayName = 'InputFiltersChip';
