// @looker/components/Form/Fields/FieldRadio/FieldRadio.js
import "core-js/modules/es.object.assign.js";

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React, { forwardRef } from 'react';
import styled from 'styled-components';
import { useID } from '../../../utils';
import { useFormContext } from '../../Form';
import { Radio } from '../../Inputs/Radio/Radio';
import { FieldInline, omitFieldProps, pickFieldProps } from '../Field';
var FieldRadioLayout = forwardRef(function (props, ref) {
  var validationMessage = useFormContext(props);
  var id = useID(props.id);
  return React.createElement(FieldInline, _extends({
    id: id,
    validationMessage: validationMessage
  }, pickFieldProps(props)), React.createElement(Radio, _extends({}, omitFieldProps(props), {
    "aria-describedby": "describedby-".concat(id),
    id: id,
    ref: ref,
    validationType: validationMessage && validationMessage.type || props.validationType
  })));
});
FieldRadioLayout.displayName = 'FieldRadioLayout';
export var FieldRadio = styled(FieldRadioLayout).withConfig({
  displayName: "FieldRadio",
  componentId: "sc-1ey14ql-0"
})([""]);
