// @looker/components/cjs/Popover/Layout/PopoverLayout.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.PopoverLayout = void 0;

var _react = _interopRequireDefault(require("react"));

var _ModalLayout = require("../../Modal/ModalLayout");

var _PopoverContent = require("./PopoverContent");

var _PopoverFooter = require("./PopoverFooter");

var _PopoverHeader = require("./PopoverHeader");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const PopoverLayout = ({
  children,
  closeButton,
  footer = true,
  header,
  hideHeader = false,
  isLoading
}) => {
  const internalFooter = typeof footer === 'boolean' ? null : footer;
  return _react.default.createElement(_react.default.Fragment, null, header && _react.default.createElement(_PopoverHeader.PopoverHeader, {
    hidden: hideHeader,
    hideClose: !!footer
  }, header), _react.default.createElement(_PopoverContent.PopoverContent, {
    hasFooter: !!footer,
    hasHeader: !!header
  }, isLoading ? _react.default.createElement(_ModalLayout.ModalLoading, null) : children), footer && _react.default.createElement(_PopoverFooter.PopoverFooter, {
    closeButton: closeButton
  }, internalFooter));
};

exports.PopoverLayout = PopoverLayout;
