// @looker/components/cjs/OrderedList/OrderedList.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.OrderedList = void 0;

var _designTokens = require("@looker/design-tokens");

var _styledComponents = _interopRequireDefault(require("styled-components"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const typeVariant = (0, _designTokens.variant)({
  prop: 'type',
  variants: {
    letter: {
      listStyleType: 'upper-alpha',
      pl: 'u4'
    },
    none: {
      listStyleType: 'none'
    },
    number: {
      listStyleType: 'decimal',
      pl: 'u4'
    }
  }
});

const OrderedList = _styledComponents.default.div.withConfig({
  shouldForwardProp: _designTokens.shouldForwardProp
}).attrs(({
  color = 'body',
  type = 'none'
}) => ({
  as: 'ol',
  color,
  type
})).withConfig({
  displayName: "OrderedList",
  componentId: "sc-csldlu-0"
})(["", " ", " ", " ", " ", " ", " ", " li{margin-bottom:", ";}"], _designTokens.reset, _designTokens.textColor, _designTokens.typography, typeVariant, _designTokens.space, _designTokens.position, _designTokens.layout, ({
  theme
}) => theme.space.u1);

exports.OrderedList = OrderedList;
