// @looker/components/ProgressCircular/DeterminateProgress.js
import "core-js/modules/es.array.concat.js";
import React from 'react';
import styled from 'styled-components';
import { CircleContainer } from './ProgressSvg';
import { progressCircularSVG } from './size';
export var DeterminateProgress = function DeterminateProgress(_ref) {
  var size = _ref.size,
      _ref$progress = _ref.progress,
      progress = _ref$progress === void 0 ? 0 : _ref$progress;

  var _progressCircularSVG = progressCircularSVG({
    size: size
  }),
      stroke = _progressCircularSVG.stroke,
      half = _progressCircularSVG.half,
      radius = _progressCircularSVG.radius,
      dashArray = _progressCircularSVG.dashArray;

  var progressOffset = (1 - progress) * (2 * Math.PI * radius);
  return React.createElement(DeterminateContainer, null, React.createElement(CircleContainer, {
    viewBox: "0 0 ".concat(half * 2, " ").concat(half * 2),
    xmlns: "http://www.w3.org/2000/svg"
  }, React.createElement(DeterminateCircle, {
    cx: half,
    cy: half,
    r: radius,
    strokeDasharray: dashArray,
    strokeDashoffset: progressOffset,
    strokeWidth: stroke
  })));
};
var DeterminateContainer = styled.div.withConfig({
  displayName: "DeterminateProgress__DeterminateContainer",
  componentId: "sc-yb0ja2-0"
})(["height:100%;position:absolute;transform:rotate(-90deg);width:100%;"]);
var DeterminateCircle = styled.circle.withConfig({
  displayName: "DeterminateProgress__DeterminateCircle",
  componentId: "sc-yb0ja2-1"
})(["stroke:", ";transition:stroke-dashoffset 500ms;"], function (_ref2) {
  var theme = _ref2.theme;
  return theme.colors.key;
});
