// @looker/components/cjs/DataTable/Filters/DataTableFilters.js
"use strict";

require("core-js/modules/es.array.index-of.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DataTableFilters = void 0;

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _react = _interopRequireDefault(require("react"));

var _designTokens = require("@looker/design-tokens");

var _Divider = require("../../Divider");

var _InputFilters = require("../../Form/Inputs/InputFilters");

var _ItemTarget = require("../Item/ItemTarget");

var _ColumnSelector = require("./ColumnSelector");

const _excluded = ["className", "children"];

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

const hasSelectableColumns = columns => Boolean(columns.find(c => c.hide !== undefined));

const DataTableFiltersLayout = _ref => {
  let className = _ref.className,
      children = _ref.children,
      props = _objectWithoutProperties(_ref, _excluded);

  const columnsSelector = _react.default.createElement(_react.default.Fragment, null, _react.default.createElement(_Divider.DividerVertical, {
    mx: "none",
    stretch: true
  }), _react.default.createElement(_ItemTarget.ItemTarget, null, _react.default.createElement(_ColumnSelector.ColumnSelector, props)));

  return _react.default.createElement("div", {
    className: className
  }, children, hasSelectableColumns(props.columns) && columnsSelector);
};

const DataTableFilters = (0, _styledComponents.default)(DataTableFiltersLayout).withConfig({
  displayName: "DataTableFilters",
  componentId: "sc-1p8hllh-0"
})(["align-items:flex-start;border-bottom:solid 1px ", ";border-top:solid 1px ", ";display:flex;", "{border:none;}", "{min-height:calc(", " - 2px);}"], ({
  theme
}) => theme.colors.ui2, ({
  theme
}) => theme.colors.ui2, _InputFilters.InputFilters, _ItemTarget.ItemTarget, _designTokens.densityTarget);
exports.DataTableFilters = DataTableFilters;
