// @looker/components/Form/Fields/FieldSelectMulti/FieldSelectMulti.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import "core-js/modules/es.array.iterator.js";
import "core-js/modules/es.object.to-string.js";
import "core-js/modules/web.dom-collections.iterator.js";
import "core-js/modules/es.object.assign.js";
import React, { forwardRef } from 'react';
import styled from 'styled-components';
import { useID } from '../../../utils';
import { useFormContext } from '../../Form';
import { SelectMulti } from '../../Inputs/Select/SelectMulti';
import { FloatingLabelField, omitFieldProps, pickFieldProps } from '../Field';

var getHasValue = function getHasValue(_ref) {
  var values = _ref.values,
      defaultValues = _ref.defaultValues;
  if (values !== undefined) return values.length > 0;
  if (defaultValues !== undefined) return defaultValues.length > 0;
  return false;
};

var checkValueOnBlur = function checkValueOnBlur(e) {
  var target = e.currentTarget;
  var options = target.querySelectorAll('[role="option"]');
  return options.length !== 0;
};

var FieldSelectMultiComponent = forwardRef(function (props, ref) {
  var validationMessage = useFormContext(props);
  var id = useID(props.id);
  return React.createElement(FloatingLabelField, _extends({
    "data-testid": "FieldSelectMultiId"
  }, pickFieldProps(props), {
    id: id,
    ariaLabelOnly: true,
    validationMessage: validationMessage,
    hasValue: getHasValue(props),
    checkValueOnBlur: checkValueOnBlur
  }), React.createElement(SelectMulti, _extends({}, omitFieldProps(props), {
    "aria-describedby": "describedby-".concat(id),
    "aria-labelledby": "labelledby-".concat(id),
    id: id,
    validationType: validationMessage && validationMessage.type,
    ref: ref
  })));
});
FieldSelectMultiComponent.displayName = 'FieldSelectMultiComponent';
export var FieldSelectMulti = styled(FieldSelectMultiComponent).withConfig({
  displayName: "FieldSelectMulti",
  componentId: "sc-ze3grr-0"
})(["width:100%;"]);
