// @looker/components/cjs/ListItem/ListItemContent.js
"use strict";

require("core-js/modules/es.array.index-of.js");

require("core-js/modules/es.array.iterator.js");

require("core-js/modules/es.weak-map.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ListItemContent = exports.listItemLabelCSS = exports.listItemContentCSS = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _utils = require("./utils");

const _excluded = ["children", "itemRole"];

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

const Button = _styledComponents.default.button.attrs(({
  type = 'button'
}) => ({
  type
})).withConfig({
  displayName: "ListItemContent__Button",
  componentId: "sc-1ietpwm-0"
})(["font-family:inherit;"]);

const Link = _styledComponents.default.a.withConfig({
  displayName: "ListItemContent__Link",
  componentId: "sc-1ietpwm-1"
})([""]);

const Div = _styledComponents.default.div.withConfig({
  displayName: "ListItemContent__Div",
  componentId: "sc-1ietpwm-2"
})([""]);

const listItemContentCSS = style => (0, _styledComponents.css)(["> ", ",> ", ",> ", "{", "}"], Button, Link, Div, style);

exports.listItemContentCSS = listItemContentCSS;
const listItemLabelCSS = listItemContentCSS;
exports.listItemLabelCSS = listItemLabelCSS;

const ListItemContentInternal = _ref => {
  let children = _ref.children,
      _ref$itemRole = _ref.itemRole,
      itemRole = _ref$itemRole === void 0 ? 'button' : _ref$itemRole,
      props = _objectWithoutProperties(_ref, _excluded);

  if (!props.disabled && itemRole === 'link') {
    return _react.default.createElement(Link, props, children);
  } else if (itemRole === 'none') {
    return _react.default.createElement(Div, props, children);
  }

  return _react.default.createElement(Button, _extends({}, props, {
    type: "button"
  }), children);
};

const ListItemContent = (0, _styledComponents.default)(ListItemContentInternal).withConfig({
  displayName: "ListItemContent",
  componentId: "sc-1ietpwm-3"
})(["", " ", " align-items:center;border:none;color:inherit;cursor:", ";display:flex;flex:1;font-size:inherit;font-weight:inherit;margin:0;min-width:0;outline:none;padding:0;", " position:relative;text-align:left;text-decoration:none;width:100%;&:hover,&:focus{color:inherit;text-decoration:none;}"], _utils.listItemBackgroundColor, ({
  focusVisible,
  theme
}) => focusVisible && `
    &::after {
      bottom: 0;
      box-shadow: inset 0 0 0 2px ${theme.colors.keyFocus};
      content: '';
      left: 0;
      position: absolute;
      right: 0;
      top: 0;
      z-index: 1;
    }
  `, ({
  cursorPointer
}) => cursorPointer ? 'pointer' : undefined, ({
  density
}) => (0, _utils.listItemPaddingX)(density));
exports.ListItemContent = ListItemContent;
