// @looker/design-tokens/cjs/legacy/palette.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.red900 = exports.red800 = exports.red700 = exports.red600 = exports.red500 = exports.red400 = exports.red300 = exports.red200 = exports.red100 = exports.red000 = exports.yellow900 = exports.yellow800 = exports.yellow700 = exports.yellow600 = exports.yellow500 = exports.yellow400 = exports.yellow300 = exports.yellow200 = exports.yellow100 = exports.yellow000 = exports.green900 = exports.green800 = exports.green700 = exports.green600 = exports.green500 = exports.green400 = exports.green300 = exports.green200 = exports.green100 = exports.green000 = exports.blue900 = exports.blue800 = exports.blue700 = exports.blue600 = exports.blue500 = exports.blue400 = exports.blue300 = exports.blue200 = exports.blue100 = exports.blue000 = exports.purple900 = exports.purple800 = exports.purple700 = exports.purple600 = exports.purple500 = exports.purple400 = exports.purple300 = exports.purple200 = exports.purple100 = exports.purple000 = exports.charcoal900 = exports.charcoal800 = exports.charcoal700 = exports.charcoal600 = exports.charcoal500 = exports.charcoal400 = exports.charcoal300 = exports.charcoal200 = exports.charcoal100 = exports.charcoal000 = exports.white = void 0;
const white = '#FFFFFF';
exports.white = white;
const charcoal000 = '#FBFBFC';
exports.charcoal000 = charcoal000;
const charcoal100 = '#F5F6F7';
exports.charcoal100 = charcoal100;
const charcoal200 = '#DEE1E5';
exports.charcoal200 = charcoal200;
const charcoal300 = '#C1C6CC';
exports.charcoal300 = charcoal300;
const charcoal400 = '#939BA5';
exports.charcoal400 = charcoal400;
const charcoal500 = '#707781';
exports.charcoal500 = charcoal500;
const charcoal600 = '#4C535B';
exports.charcoal600 = charcoal600;
const charcoal700 = '#343C42';
exports.charcoal700 = charcoal700;
const charcoal800 = '#262D33';
exports.charcoal800 = charcoal800;
const charcoal900 = '#262D33';
exports.charcoal900 = charcoal900;
const purple000 = '#F3F2FF';
exports.purple000 = purple000;
const purple100 = '#E8E5FF';
exports.purple100 = purple100;
const purple200 = '#BFB2FF';
exports.purple200 = purple200;
const purple300 = '#9785F2';
exports.purple300 = purple300;
const purple400 = '#6C43E0';
exports.purple400 = purple400;
const purple500 = '#4F2ABA';
exports.purple500 = purple500;
const purple600 = '#412399';
exports.purple600 = purple600;
const purple700 = '#341C7A';
exports.purple700 = purple700;
const purple800 = '#291661';
exports.purple800 = purple800;
const purple900 = '#1E1047';
exports.purple900 = purple900;
const blue000 = '#f7fcff';
exports.blue000 = blue000;
const blue100 = '#e0f2ff';
exports.blue100 = blue100;
const blue200 = '#bfe3ff';
exports.blue200 = blue200;
const blue300 = '#6fbcf7';
exports.blue300 = blue300;
const blue400 = '#49a9f2';
exports.blue400 = blue400;
const blue500 = '#0087e1';
exports.blue500 = blue500;
const blue600 = '#0059b2';
exports.blue600 = blue600;
const blue700 = '#00418c';
exports.blue700 = blue700;
const blue800 = '#0f2f66';
exports.blue800 = blue800;
const blue900 = '#08284d';
exports.blue900 = blue900;
const green000 = '#f2fff5';
exports.green000 = green000;
const green100 = '#e0ffe7';
exports.green100 = green100;
const green200 = '#b4fac2';
exports.green200 = green200;
const green300 = '#67e591';
exports.green300 = green300;
const green400 = '#33cc73';
exports.green400 = green400;
const green500 = '#24b25f';
exports.green500 = green500;
const green600 = '#0e8c42';
exports.green600 = green600;
const green700 = '#0b6b38';
exports.green700 = green700;
const green800 = '#08522d';
exports.green800 = green800;
const green900 = '#06381f';
exports.green900 = green900;
const yellow000 = '#FFF7E8';
exports.yellow000 = yellow000;
const yellow100 = '#FFEBC4';
exports.yellow100 = yellow100;
const yellow200 = '#FFDB96';
exports.yellow200 = yellow200;
const yellow300 = '#FFCA62';
exports.yellow300 = yellow300;
const yellow400 = '#FFB72B';
exports.yellow400 = yellow400;
const yellow500 = '#FFA800';
exports.yellow500 = yellow500;
const yellow600 = '#EF9E00';
exports.yellow600 = yellow600;
const yellow700 = '#CC8600';
exports.yellow700 = yellow700;
const yellow800 = '#A56D00';
exports.yellow800 = yellow800;
const yellow900 = '#724B00';
exports.yellow900 = yellow900;
const red000 = '#FFF2F4';
exports.red000 = red000;
const red100 = '#FFE5E9';
exports.red100 = red100;
const red200 = '#FFB8C1';
exports.red200 = red200;
const red300 = '#FF667A';
exports.red300 = red300;
const red400 = '#ED3B53';
exports.red400 = red400;
const red500 = '#CC1F36';
exports.red500 = red500;
const red600 = '#B2121F';
exports.red600 = red600;
const red700 = '#990F14';
exports.red700 = red700;
const red800 = '#730B0F';
exports.red800 = red800;
const red900 = '#52080B';
exports.red900 = red900;
