// @looker/components/cjs/Form/Fields/Field/Field.js
"use strict";

require("core-js/modules/es.weak-map.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Field = exports.omitFieldProps = exports.pickFieldProps = exports.fieldPropKeys = void 0;

require("core-js/modules/es.array.iterator.js");

require("core-js/modules/es.symbol.description.js");

var _designTokens = require("@looker/design-tokens");

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _omit = _interopRequireDefault(require("lodash/omit"));

var _pick = _interopRequireDefault(require("lodash/pick"));

var _height = require("../../Inputs/height");

var _HelperText = require("./HelperText");

var _FieldDetail = require("./FieldDetail");

var _FieldLabel = require("./FieldLabel");

var _InputArea = require("./InputArea");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const fieldPropKeys = ['className', 'description', 'detail', 'externalLabel', 'id', 'inline', 'label', 'hideLabel', 'labelWidth', 'validationMessage', 'width'];
exports.fieldPropKeys = fieldPropKeys;

const pickFieldProps = props => (0, _pick.default)(props, [...fieldPropKeys, 'disabled', 'required', 'autoResize']);

exports.pickFieldProps = pickFieldProps;

const omitFieldProps = props => (0, _omit.default)(props, fieldPropKeys);

exports.omitFieldProps = omitFieldProps;
const Field = (0, _styledComponents.default)(({
  autoResize,
  className,
  children,
  description,
  detail,
  id,
  ariaLabelOnly,
  label,
  hideLabel,
  required,
  validationMessage
}) => _react.default.createElement("div", {
  className: className
}, _react.default.createElement(_FieldLabel.FieldLabel, {
  ariaLabelOnly: ariaLabelOnly,
  id: id,
  label: label,
  hideLabel: hideLabel,
  required: required
}), _react.default.createElement(_InputArea.InputArea, {
  autoResize: autoResize
}, children), detail && _react.default.createElement(_FieldDetail.FieldDetail, null, detail), _react.default.createElement(_HelperText.HelperText, {
  description: description,
  id: id,
  validationMessage: validationMessage
}))).withConfig({
  displayName: "Field",
  componentId: "sc-1luvvdx-0"
})(["align-items:left;display:", ";", " grid-template-columns:", ";height:fit-content;justify-content:space-between;width:", ";", " ", "{grid-area:input;}", "{grid-area:messages;}& > ", "{grid-area:label;", "}", "{grid-area:detail;justify-self:end;padding-left:", ";", "}"], ({
  autoResize
}) => autoResize ? 'inline-grid' : 'grid', ({
  inline
}) => inline ? inlineTemplateAreas : templateAreas, ({
  inline
}) => inline ? '150px 1fr' : undefined, ({
  autoResize
}) => autoResize ? 'fit-content' : '100%', _designTokens.width, _InputArea.InputArea, _HelperText.HelperText, _FieldLabel.FieldLabel, ({
  inline
}) => fieldLabelCSS(inline), _FieldDetail.FieldDetail, ({
  theme: {
    space
  }
}) => space.u3, ({
  inline
}) => inline && (0, _styledComponents.css)(["align-self:center;"]));
exports.Field = Field;

const fieldLabelCSS = inline => inline ? (0, _styledComponents.css)(["height:", ";justify-self:end;line-height:", ";padding-right:", ";text-align:right;"], _height.inputHeight, _height.inputHeight, ({
  theme
}) => theme.space.u3) : (0, _styledComponents.css)(["line-height:", ";padding-bottom:", ";"], ({
  theme
}) => theme.lineHeights.xsmall, ({
  theme
}) => theme.space.u1);

const inlineTemplateAreas = (0, _styledComponents.css)(["grid-template-areas:'label input detail' '. messages messages';"]);
const templateAreas = (0, _styledComponents.css)(["grid-template-areas:'label detail' 'input input' 'messages messages';"]);
