// @looker/components/cjs/Form/Fields/Field/HelperText.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.HelperText = void 0;

require("core-js/modules/es.symbol.description.js");

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _Layout = require("../../../Layout");

var _Text = require("../../../Text");

var _ValidationMessage = require("../../ValidationMessage");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const HelperText = (0, _styledComponents.default)(({
  className,
  description,
  id,
  validationMessage
}) => {
  return _react.default.createElement(_Layout.SpaceVertical, {
    pt: description || validationMessage ? 'u2' : 'none',
    gap: "u1",
    className: className,
    id: `describedby-${id}`
  }, description && _react.default.createElement(_Text.Paragraph, {
    fontSize: "xsmall",
    color: "text2"
  }, description), validationMessage && _react.default.createElement(_ValidationMessage.ValidationMessage, validationMessage));
}).withConfig({
  displayName: "HelperText",
  componentId: "sc-45tbno-0"
})([""]);
exports.HelperText = HelperText;
