// @looker/components/Density/Density.js
import "core-js/modules/es.object.assign.js";

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React, { useContext } from 'react';
import { ThemeContext, ThemeProvider } from 'styled-components';
export var Density = function Density(_ref) {
  var scale = _ref.scale,
      children = _ref.children;
  var theme = useContext(ThemeContext);
  theme.defaults.density = scale;
  return React.createElement(ThemeProvider, {
    theme: theme
  }, children);
};
export var Density1 = function Density1(props) {
  return React.createElement(Density, _extends({
    scale: 1
  }, props));
};
export var Density0 = function Density0(props) {
  return React.createElement(Density, _extends({
    scale: 0
  }, props));
};
export var DensityNegative1 = function DensityNegative1(props) {
  return React.createElement(Density, _extends({
    scale: -1
  }, props));
};
export var DensityNegative2 = function DensityNegative2(props) {
  return React.createElement(Density, _extends({
    scale: -2
  }, props));
};
export var DensityNegative3 = function DensityNegative3(props) {
  return React.createElement(Density, _extends({
    scale: -3
  }, props));
};
