// @looker/components/Link/Link.js
var _excluded = ["children", "isExternal"];

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _toConsumableArray(arr) { return _arrayWithoutHoles(arr) || _iterableToArray(arr) || _unsupportedIterableToArray(arr) || _nonIterableSpread(); }

function _nonIterableSpread() { throw new TypeError("Invalid attempt to spread non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _iterableToArray(iter) { if (typeof Symbol !== "undefined" && iter[Symbol.iterator] != null || iter["@@iterator"] != null) return Array.from(iter); }

function _arrayWithoutHoles(arr) { if (Array.isArray(arr)) return _arrayLikeToArray(arr); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

import "core-js/modules/es.regexp.exec.js";
import "core-js/modules/es.string.split.js";
import "core-js/modules/es.array.join.js";
import "core-js/modules/es.array.iterator.js";
import "core-js/modules/es.object.to-string.js";
import "core-js/modules/es.set.js";
import "core-js/modules/es.string.iterator.js";
import "core-js/modules/web.dom-collections.iterator.js";
import "core-js/modules/es.array.concat.js";
import "core-js/modules/es.string.link.js";
import "core-js/modules/es.array.is-array.js";
import "core-js/modules/es.symbol.js";
import "core-js/modules/es.symbol.description.js";
import "core-js/modules/es.symbol.iterator.js";
import "core-js/modules/es.array.from.js";
import "core-js/modules/es.array.slice.js";
import "core-js/modules/es.function.name.js";
import "core-js/modules/es.object.keys.js";
import "core-js/modules/es.array.index-of.js";
import "core-js/modules/es.object.assign.js";
import { reset, typography, omitStyledProps } from '@looker/design-tokens';
import { Launch } from '@styled-icons/material/Launch';
import omit from 'lodash/omit';
import styled from 'styled-components';
import React, { forwardRef } from 'react';
var ExternalLinkIndicator = styled(Launch).withConfig({
  displayName: "Link__ExternalLinkIndicator",
  componentId: "sc-165dqum-0"
})(["height:", ";margin-left:", ";width:", ";"], function (_ref) {
  var theme = _ref.theme;
  return theme.sizes.xxsmall;
}, function (_ref2) {
  var theme = _ref2.theme;
  return theme.space.u1;
}, function (_ref3) {
  var theme = _ref3.theme;
  return theme.sizes.xxsmall;
});

var generateLinkTypes = function generateLinkTypes(_ref4) {
  var dangerouslyDisableRel = _ref4.dangerouslyDisableRel,
      isExternal = _ref4.isExternal,
      rel = _ref4.rel,
      target = _ref4.target;
  if (dangerouslyDisableRel) return rel;
  var linkTypes = rel ? rel.split(' ') : [];

  if (target === '_blank') {
    linkTypes.push('noopener', 'noreferrer');
  } else if (isExternal) {
    linkTypes.push('external', 'noreferrer');
  }

  return _toConsumableArray(new Set(linkTypes)).join(' ');
};

var linkStyleProps = ['keyColor', 'underline'];
var LinkLayout = forwardRef(function (props, ref) {
  var children = props.children,
      isExternal = props.isExternal,
      restProps = _objectWithoutProperties(props, _excluded);

  var enhancedRel = generateLinkTypes(props);
  return React.createElement("a", _extends({}, omit(omitStyledProps(restProps), [].concat(linkStyleProps, ['dangerouslyDisableRel'])), {
    ref: ref,
    rel: enhancedRel
  }), children, isExternal && React.createElement(ExternalLinkIndicator, null));
});
LinkLayout.displayName = 'LinkLayout';
export var Link = styled(LinkLayout).withConfig({
  displayName: "Link",
  componentId: "sc-165dqum-1"
})(["", " ", " color:", ";text-decoration:", ";&[aria-expanded='true'],&:focus,&:hover,&:active,&.active,&:visited{color:", ";outline:none;text-decoration:underline;}"], reset, typography, function (_ref5) {
  var keyColor = _ref5.keyColor,
      colors = _ref5.theme.colors;
  return keyColor ? colors.key : colors.link;
}, function (_ref6) {
  var underline = _ref6.underline;
  return underline ? 'underline' : 'none';
}, function (_ref7) {
  var keyColor = _ref7.keyColor,
      colors = _ref7.theme.colors;
  return keyColor ? colors.keyInteractive : colors.linkInteractive;
});
