// @looker/components/Form/Inputs/InputColor/Handle/Handle.js
import "core-js/modules/es.string.small.js";
import "core-js/modules/es.array.concat.js";
import { theme } from '@looker/design-tokens';
import styled, { css } from 'styled-components';
export var HANDLE_SIZE = theme.sizes.small;
export var handleCSS = css(["border:2px solid ", ";border-radius:100%;box-shadow:", ";cursor:", ";height:", ";left:0;position:relative;width:", ";"], function (_ref) {
  var colors = _ref.theme.colors;
  return colors.background;
}, function (_ref2) {
  var theme = _ref2.theme;
  return theme.elevations.plus1;
}, function (_ref3) {
  var isMouseDown = _ref3.isMouseDown;
  return isMouseDown ? 'grabbing' : 'pointer';
}, HANDLE_SIZE, HANDLE_SIZE);
export var Handle = styled.div.attrs(function (_ref4) {
  var color = _ref4.color,
      x = _ref4.x;
  return {
    style: {
      background: color,
      transform: "translateX(calc(".concat(x, "px - ").concat(HANDLE_SIZE, " / 2))")
    }
  };
}).withConfig({
  displayName: "Handle",
  componentId: "sc-1ojot3e-0"
})(["", " top:", ";"], handleCSS, function (_ref5) {
  var theme = _ref5.theme;
  return "calc(".concat(theme.space.u3, " / 2 - ").concat(HANDLE_SIZE, " / 2)");
});
