// @looker/components/cjs/Avatar/AvatarCombo.js
"use strict";

require("core-js/modules/es.array.index-of.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.AvatarCombo = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _designTokens = require("@looker/design-tokens");

var _PersonOutline = require("@styled-icons/material/PersonOutline");

var _styledIcon = require("@styled-icons/styled-icon");

var _AvatarUser = require("./AvatarUser");

var _AvatarIcon = require("./AvatarIcon");

var _Avatar = require("./Avatar");

const _excluded = ["secondaryIcon", "secondaryColor", "secondaryBg", "color", "icon", "user", "role"];

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

const AvatarIconSecondary = (0, _styledComponents.default)(_AvatarIcon.AvatarIcon).withConfig({
  displayName: "AvatarCombo__AvatarIconSecondary",
  componentId: "sc-gekgjm-0"
})([""]);

const AvatarLayout = _ref => {
  let secondaryIcon = _ref.secondaryIcon,
      secondaryColor = _ref.secondaryColor,
      secondaryBg = _ref.secondaryBg,
      color = _ref.color,
      _ref$icon = _ref.icon,
      icon = _ref$icon === void 0 ? _react.default.createElement(_PersonOutline.PersonOutline, null) : _ref$icon,
      user = _ref.user,
      role = _ref.role,
      props = _objectWithoutProperties(_ref, _excluded);

  const BaseElement = role === 'button' ? 'button' : 'div';
  return _react.default.createElement(BaseElement, (0, _designTokens.omitStyledProps)(props), user ? _react.default.createElement(_AvatarUser.AvatarUser, {
    user: user,
    color: color
  }) : _react.default.createElement(_AvatarIcon.AvatarIcon, {
    color: color,
    icon: icon
  }), _react.default.createElement(AvatarIconSecondary, {
    bg: secondaryBg,
    color: secondaryColor,
    icon: secondaryIcon
  }));
};

const AvatarCombo = (0, _styledComponents.default)(AvatarLayout).withConfig({
  displayName: "AvatarCombo",
  componentId: "sc-gekgjm-1"
})(["", " border:none;height:40px;position:relative;width:40px;", "{bottom:-4px;height:20px;position:absolute;right:-4px;width:20px;", "{height:14px;width:14px;}}"], _Avatar.avatarButtonOverrides, AvatarIconSecondary, _styledIcon.StyledIconBase);
exports.AvatarCombo = AvatarCombo;
