// @looker/components/cjs/Avatar/AvatarUser.js
"use strict";

require("core-js/modules/es.array.index-of.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.AvatarUser = void 0;

var _reactI18next = require("react-i18next");

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _designTokens = require("@looker/design-tokens");

var _Avatar = require("./Avatar");

const _excluded = ["color", "user", "role", "size"];

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

const AvatarLayout = _ref => {
  let color = _ref.color,
      user = _ref.user,
      role = _ref.role,
      size = _ref.size,
      props = _objectWithoutProperties(_ref, _excluded);

  const _useTranslation = (0, _reactI18next.useTranslation)('AvatarUser'),
        t = _useTranslation.t;

  const firstInitial = user && user.first_name && user.first_name[0];
  const lastInitial = user && user.last_name && user.last_name[0];
  const name = user ? `${user.first_name} ${user.last_name}` : t('Avatar');
  const BaseElement = role === 'button' ? 'button' : 'div';
  return _react.default.createElement(BaseElement, _extends({}, (0, _designTokens.omitStyledProps)(props), {
    "aria-label": name
  }), _react.default.createElement(AvatarInitials, {
    color: color,
    "aria-hidden": true
  }, size === 'xxsmall' ? `${firstInitial}` : `${firstInitial}${lastInitial}`), user && user.avatar_url && _react.default.createElement(AvatarPhoto, {
    "aria-hidden": true,
    "data-testid": "avatar-photo",
    color: color,
    type: "image/png",
    data: user.avatar_url
  }));
};

const AvatarPhoto = _styledComponents.default.object.withConfig({
  displayName: "AvatarUser__AvatarPhoto",
  componentId: "sc-ds3q2d-0"
})(["bottom:0;height:100%;left:0;position:absolute;right:0;top:0;width:100%;"]);

const AvatarInitials = _styledComponents.default.div.withConfig({
  shouldForwardProp: _designTokens.shouldForwardProp
}).attrs(({
  color
}) => ({
  bg: color
})).withConfig({
  displayName: "AvatarUser__AvatarInitials",
  componentId: "sc-ds3q2d-1"
})(["color:", ";"], ({
  theme
}) => theme.colors.keyText);

const AvatarUser = (0, _styledComponents.default)(AvatarLayout).attrs(({
  color = 'key',
  size = 'small'
}) => ({
  color,
  size
})).withConfig({
  displayName: "AvatarUser",
  componentId: "sc-ds3q2d-2"
})(["", " ", " background:currentColor;position:relative;"], _Avatar.avatarCSS, ({
  role
}) => role === 'button' && 'cursor: pointer;');
exports.AvatarUser = AvatarUser;
