// @looker/components/cjs/Avatar/Avatar.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.avatarCSS = exports.avatarButtonOverrides = void 0;

var _designTokens = require("@looker/design-tokens");

var _styledComponents = require("styled-components");

const size = (0, _designTokens.variant)({
  prop: 'size',
  variants: {
    xxsmall: {
      fontSize: 'xsmall',
      height: '24px',
      width: '24px'
    },
    xsmall: {
      fontSize: 'xsmall',
      height: '32px',
      width: '32px'
    },
    small: {
      fontSize: 'small',
      height: '40px',
      width: '40px'
    },
    medium: {
      fontSize: 'medium',
      height: '54px',
      width: '54px'
    },
    large: {
      fontSize: 'large',
      height: '60px',
      width: '60px'
    }
  }
});
const avatarButtonOverrides = (0, _styledComponents.css)(["background:transparent;border:none;padding:0;"]);
exports.avatarButtonOverrides = avatarButtonOverrides;
const avatarCSS = (0, _styledComponents.css)(["", " ", " ", " ", " ", " ", " align-items:center;border-radius:100%;display:grid;flex-shrink:0;justify-items:center;overflow:hidden;&:focus{outline:none;}"], _designTokens.reset, avatarButtonOverrides, _designTokens.color, _designTokens.space, _designTokens.typography, size);
exports.avatarCSS = avatarCSS;
